sympy/core/mul.py

```diff
@@ -1250,6 +1250,26 @@ def _eval_is_zero(self):
         return zero
 
     def _eval_is_integer(self):
+        if len(self.args) == 2:
+            a, b = self.args
+            if None not in (a.is_integer, b.is_integer):
+                if b.is_integer:
+                    a, b = b, a
+                if a.is_integer:
+                    if b.is_integer:
+                        # both ints
+                        return True
+                    if a.is_zero and b.is_finite or b.is_zero:
+                        return True
+                    if a.is_zero is False and b.is_zero is False:
+                        if b.is_real is False:
+                            return False
+                        if b.is_rational is False or (
+                                a.is_even != (1/b).is_even and
+                                None not in (
+                                a.is_even, (1/b).is_even)):
+                            return False
+
```

Comment:
I think more could be done for the rational case:
```python
t = ne, de = n.is_even, d.is_even
if None not in t:
    if ne is True and de is True:
        return True
    if ne != de:
        return False
    # both odd
    if n.is_prime and d.is_prime:
        return False
    # don't know

Reply:
I'll read up on a couple properties, try to implement them, and check them 👍

Reply:
Researching what? If you are responding to the "don't know" I just mean that the routine can't tell in this case so nothing is being returned. A division of two odd numbers that are not both prime will only be an integer if they share a common divisor...which we can't test symbolically.

Reply:
Ohhh that part ... I was thinking more like ... trying out cases
```python3
>>> Mul(Rational(1,8), Rational(8,1), evaluate=False)
8/8
>>> Mul(Rational(1,8), Rational(8,1), evaluate=False).is_integer
>>> 
```
for situations like these ... 
> Researching what?

I just plan on going through various cases ... 

Like 1/8 terminates after 3 dec places, it's a lot easier to work with than 1/3
```python3
>>> sympify(1/8)
0.125000000000000
>>> Rational(_)
1/8
>>> sympify(1/3)
0.333333333333333
>>> Rational(_)
6004799503160661/18014398509481984
>>> 
```

Reply:
Use S(1)/3 to get a Rational 1/3 instead of a Float (or a Rational approximation of that Float).